lib/active_model/serializer/associations.rb

```diff
@@ -74,7 +74,8 @@ def has_one(name, options = {}, &block) # rubocop:disable Style/PredicateName
         # @api private
         #
         def associate(reflection)
-          self._reflections << reflection
+          key = reflection.options[:key]
+          key ? self._reflections[key] = reflection : self._reflections[reflection.name] = reflection
```

Comment:
Because you can't handle some situations solely with name i.e:
`has_many :comments
  has_many :comments, key: :last_comments do
     object.comments.last(1)
   end`